galaxyproject/galaxy
# Restore homogeneous dataset collection indicator

Fixes #13744

<del>I had to move the `elements_datatype` serialization field from the `detailed` to the `summary` view so it can be rendered in the history items without making additional requests. Hope that's ok.</del> See https://github.com/galaxyproject/galaxy/pull/13772#issuecomment-1106246781

Also, I slightly changed the description string (compared to the old history) depending on the `collection_type`, for example:
`list:paired` (heterogeneous): *a list of pairs with 10 items*  -> *a list with 10 pairs*
`list:paired` (homogeneous): *a list of pairs with 10 items*  -> *a list with 10 **tabular** pairs*

There is a complete [list of samples in the tests](https://github.com/galaxyproject/galaxy/compare/dev...davelopez:restore_homogeneous_hdca_indicator?expand=1#diff-a68e5baf968484751bfe46e55652d7ae3fa46c57a4abd21b08d5a8f3abcec32eR19-R58). I'm happy to further tweak this if neccesary.

## How to test the changes?
  - Create some collections with the same (and different) types of datasets
  - Observe the collection description matches the contents

![Screenshot from 2022-04-20 14-55-55](https://user-images.githubusercontent.com/46503462/164240005-66d6c678-5a4e-4334-9033-c3e62cc1a44a.png)

## License
